### PR TITLE
Sync controller media sequence strategy

### PR DESCRIPTION
**Pull request by dzianis-dashkevich**
- implement a new sync strategy based on the media sequence
- implement exact match for sync controller
- fix fast quality switch race condition
- implement force timestamp offset after resync loader

### CODE REVIEW

File: src/segment-loader.js [hunk 1]
```diff
@@ -1433,6 +1437,7 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     if (next.isSyncRequest) {
       next.mediaIndex = getSyncSegmentCandidate(this.currentTimeline_, segments, bufferedEnd);
+      this.logger_(`choose next request. Cant find sync point. Fallback to media Index: ${next.mediaIndex}`);
```
Comment (wseymour15) [inline src/segment-loader.js:1440]: nit sp: `Can't` or 
```suggestion
      this.logger_(`choose next request. Can not find sync point. Fallback to media Index: ${next.mediaIndex}`);
```

**Review by wseymour15 — APPROVED** [review]
These changes make sense to me! Approving under the assumption that we are coming back to this and cleaning up the skipped tests

**Review by adrums86 — APPROVED** [review]
LGTM. I think the skips are okay here as these tests seem to be dependent on the old sync strategies throwing away a request. I do however think we should add some new sync-controller tests for the new strategy, either in this PR or a follow up.

**Comment by dzianis-dashkevich** [comment]
@adrums86, I will add it in the next PRs

**Pull request merged**